Automattic/wp-calypso
# state(media): add error handling for updating/deleting media items

#### Changes proposed in this Pull Request

It seems like we didn't have any error handling in place for editing and deleting media items.

- Add basic error handling for editing and deleting media items
- Remove flux actions from those data layer handlers as they weren't doing anything anymore

#### Testing instructions

Depending on your browser to simulate failures when processing media files like mentioned above enable request blocking for the following URLs:

- deleting: `https://public-api.wordpress.com/rest/v1.1/sites/[YOUR-SITE-ID]/media/*/delete`
- editing: `https://public-api.wordpress.com/rest/v1.1/sites/[YOUR-SITE-ID]/media/*/edit`

Then open the media library and try to edit a media item (e.g. rotate, flip or crop an item) and delete a media item. In both cases you should see an error notice appearing in the top right which says it couldn't handle the corresponding action.

<img width="447" alt="Screenshot 2020-10-26 at 17 38 52" src="https://user-images.githubusercontent.com/9202899/97200910-1e020b00-17b2-11eb-8927-68527eaaed2c.png">
